src/Components/TodoContext/TodoContext.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { getTodos } from '../../api/todos';
+
+export enum FilterSettings {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
+
+type Props = {
+  children: React.ReactNode;
+};
+
+type TodoContextType = {
+  todosList: Todo[];
+  setTodosList: (v: Todo[]) => void;
+
+  newTodo: Todo;
+  setNewTodo: (v: {
+    title: string;
+    id: number;
+    completed: boolean;
+    userId: number;
+  }) => void;
+
+  filterSettings: FilterSettings;
+  setFilterSettings: (v: FilterSettings) => void;
+
+  errorMessage: string;
+  setErrorMessage: (v: string) => void;
+
+  newTodoProcessing: boolean;
+  setNewTodoProcessing: (v: boolean) => void;
+};
+
+export const todoPattern = {
+  title: '',
+  id: +new Date(),
+  completed: false,
+  userId: 501,
+};
+
+export const TodoContext = React.createContext<TodoContextType>({
+  todosList: [],
+  setTodosList: () => {},
+  newTodo: todoPattern,
```

Comment:
no need to store the entire object - just title (string). It can be defined in the `Header` component